clouddriver-kubernetes-v2/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/job/KubernetesRunJobOperationTest.java

```diff
@@ -0,0 +1,169 @@
+import com.netflix.spinnaker.clouddriver.security.ProviderVersion;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+@RunWith(JUnitPlatform.class)
+final class KubernetesRunJobOperationTest {
+  private static final String NAMESPACE = "my-namespace";
+  private static final String GENERATE_SUFFIX = "-abcd";
+  private static final ResourcePropertyRegistry resourcePropertyRegistry =
+      new GlobalResourcePropertyRegistry(
+          ImmutableList.of(new KubernetesReplicaSetHandler(), new KubernetesServiceHandler()),
+          new KubernetesUnregisteredCustomResourceHandler());
+
+  @BeforeEach
+  void setTask() {
+    TaskRepository.threadLocalTask.set(new DefaultTask("task-id"));
+  }
+
+  @Test
+  void deploysJobWithName() {
+    KubernetesRunJobOperationDescription runJobDescription = baseJobDescription("job.yml");
+    OperationResult result = operate(runJobDescription);
+
+    assertThat(result.getManifestNamesByNamespace()).containsOnlyKeys(NAMESPACE);
+    assertThat(result.getManifestNamesByNamespace().get(NAMESPACE))
+        .containsExactlyInAnyOrder("job my-job");
+  }
+
+  @Test
+  void deploysJobWithGenerateName() {
+    KubernetesRunJobOperationDescription runJobDescription =
+        baseJobDescription("job-generate-name.yml");
+    OperationResult result = operate(runJobDescription);
+
+    assertThat(result.getManifestNamesByNamespace()).containsOnlyKeys(NAMESPACE);
+    assertThat(result.getManifestNamesByNamespace().get(NAMESPACE))
+        .containsExactlyInAnyOrder("job my-job" + GENERATE_SUFFIX);
+  }
+
+  @Test
+  void overridesNamespace() {
+    String overrideNamespace = "override-namespace";
+    KubernetesRunJobOperationDescription runJobDescription =
+        baseJobDescription("job.yml").setNamespace(overrideNamespace);
+    OperationResult result = operate(runJobDescription);
+
+    assertThat(result.getManifestNamesByNamespace()).containsOnlyKeys(overrideNamespace);
+    assertThat(result.getManifestNamesByNamespace().get(overrideNamespace))
+        .containsExactlyInAnyOrder("job my-job");
+  }
+
+  private static KubernetesRunJobOperationDescription baseJobDescription(String manifest) {
+    KubernetesRunJobOperationDescription runJobDescription =
+        new KubernetesRunJobOperationDescription()
+            .setManifest(
+                ManifestFetcher.getManifest(KubernetesRunJobOperationTest.class, manifest));
+    runJobDescription.setCredentials(getNamedAccountCredentials());
+    return runJobDescription;
+  }
+
+  private static KubernetesNamedAccountCredentials<KubernetesV2Credentials>
+      getNamedAccountCredentials() {
+    KubernetesConfigurationProperties.ManagedAccount managedAccount =
+        new KubernetesConfigurationProperties.ManagedAccount();
+    managedAccount.setName("my-account");
+    managedAccount.setProviderVersion(ProviderVersion.v2);
```

Comment:
I think v2 is the [default](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/config/KubernetesConfigurationProperties.java#L32) now so you might not need this line!

Reply:
You are entirely correct, removed it! 🎉